docs/ROOTFS_DISCOVERY.md

```diff
@@ -0,0 +1,395 @@
+   used in a very restricted fashion, i.e. configure some UI details as well as
+   menu entries. Some options available are ignored if Secure Boot mode is
+   enabled. Moreover, even if the text strings shown in the menu entries might
+   not be authenticated the binaries that are invoked once they are selected
+   are.
+
+2. The UKIs are also authenticated by the firmware via UEFI Secure Boot, and so
+   are EFI Addons. `confext` and `sysext` sidecars are protected via
+   `dm-verity` and a signature of the root hash is validated against keys in
+   the kernel's built-in keyring. System credentials are authenticated via
+   secrets stored in the TPM.
+
+3. Authentication of the `rootfs` and `usrfs` is more variable: depending on
+   setup this is either done via `dm-verity` (either pinned by root hash from
+   the UKI, or authenticated by signature prvoided to kernel, checked against
+   the kernel's built-in keyring) or `dm-crypt`+`dm-integrity` (protected by
+   TPM or user provided password/FIDO/PKCS#11), or in the network case at
+   download time via detached signatures (currently only GPG) or via HTTPS
+   certificate validation. Note that by default the automatic discovery
+   mechanism of the `rootfs` and its auxiliary file systems does not insist on
+   cryptographic protection and authentication before use. However, the
+   [`systemd.image_policy=`](https://www.freedesktop.org/software/systemd/man/latest/systemd.image-policy.html)
+   kernel command line switch may be used to control precisely what kind of
+   protection to require from each such partition.
+
+Note that UEFI Secure Boot is problematic in various ways: it is generally
+bound to a certificate list maintained centrally by Microsoft, and thus implies
+a complex (and expensive) code signing bureaucracy, that in many cases is
+undesirable, particularly in a community Linux world. Moreover, because he
+certficate list is very large its security value is limited: it mostly acts
+more as denylist of known bad software rather than as allowlist of known good.
+
+### Shim
+
+To make the code signing more palatable to the Linux world the `shim` project
+has been developed, which is often used as initial component of the OS boot
+(i.e. the firmware would invoke `shim` as component 0, before the components 1,
+2, 3 described above). `shim` is primarily relevant for two reasons: it's adds
+a second set of certificates on top of the UEFI Secure Boot list, maintained by
+the OS vendor, and it optionally provides functionality to maintain a local set
+of keys ("MOK") in addition to the Microsoft and OS vendor keys.
+
+### Automatic Enrollment of Secure Boot Certificates
+
+The `systemd-boot` boot loader also supports automatic enrollment of alernative
+SecureBoot certificates: if the system is booted in the firmware-provided
+Secure Boot "Setup Mode", it can automatically enroll certificates placed
+inside the ESP into the firmware, replacing the Microsoft provided certificate
```

Comment:
I think saying "replacing" here is wrong, there might be no Microsoft certificates in place at all. Just say that setup mode allows you to enroll new certificates, replacing any existing ones if there are any.